android/src/main/kotlin/com/kiwi/reversegeocode/RNReverseGeocodeManager.kt

```diff
@@ -104,19 +104,29 @@ class RNReverseGeocodeManager (reactContext: ReactApplicationContext) : ReactCon
         upperRightLatitude,
         upperRightLongitude
       )
-
-      var remoteAddressesMaxResults = MAX_RESULTS;
-      if (localAddresses != null) {
-        remoteAddressesMaxResults = MAX_RESULTS - localAddresses.size
-      } 
-
-      // Then add other more remote possibilities
-      val remoteAddresses = geocoder.getFromLocationName(
-        searchText,
-        remoteAddressesMaxResults
-      )
-
-      val addresses = remoteAddresses.orEmpty() + localAddresses.orEmpty()
+      var addresses = localAddresses.orEmpty()
+
+      // If less than the max-amount of items have been found,
+      // then add other more remote possibilities
+      if ((localAddresses != null) && (localAddresses.size < MAX_RESULTS)) {
+        var remoteAddresses = geocoder.getFromLocationName(
+          searchText,
+          MAX_RESULTS
+        )
+        if (remoteAddresses != null) {
+
+          // Filter out local-search duplicates
+          remoteAddresses = remoteAddresses.filter { ra -> localAddresses.find { la -> la.getLatitude() == ra.getLatitude() && la.getLongitude() == ra.getLongitude() } == null };
```

Comment:
Could you rename `ra` to `remoteAddress` and `la` to `localAddress`, please? I know it is more verbose, but I think it reads better.